core/asm/asm.go

```diff
@@ -0,0 +1,138 @@
+// Copyright 2017 The go-ethereum Authors
+// This file is part of the go-ethereum library.
+//
+// The go-ethereum library is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Lesser General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// The go-ethereum library is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU Lesser General Public License for more details.
+//
+// You should have received a copy of the GNU Lesser General Public License
+// along with the go-ethereum library. If not, see <http://www.gnu.org/licenses/>.
+
+// Provides support for dealing with EVM assembly instructions (e.g., disassembling them).
+package asm
+
+import (
+	"encoding/hex"
+	"fmt"
+
+	"github.com/ethereum/go-ethereum/core/vm"
+)
+
+// Iterator for disassembled EVM instructions
+type instructionIterator struct {
+	code  []byte
+	pc    uint64
+	arg   []byte
+	op    vm.OpCode
+	error error
+}
+
+// Create a new instruction iterator.
+func NewInstructionIterator(code []byte) *instructionIterator {
+	it := new(instructionIterator)
+	it.code = code
+	it.pc = 0
```

Comment:
There's no need to initialise `pc`, `error`, `arg`. These are by default initialised to their default value i.e. `0` and `nil`.